src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useContext } from 'react';
+import * as todoService from '../../api/todos';
+import { Filter } from '../Filter';
+import { TodoContext } from '../../context/TodoContext';
+import { Errors } from '../../types/Errors';
+import { Todo } from '../../types/Todo';
+
+export const Footer: React.FC = () => {
+  const { todos, setTodos, setMultiLoader, setErrorMessage, titleField } =
+    useContext(TodoContext);
+
+  const hasEnoughTodos = todos.length > 0;
+
+  const uncompletedTodos = todos.filter(todo => todo.completed === false);
+
+  const hasEnoughCompletedTodo = todos.some(todo => todo.completed === true);
+
+  const removeTodos = (completedTodos: Todo[]) => {
+    completedTodos.forEach(todo => {
+      setMultiLoader(true);
+      todoService
+        .deleteTodo(todo.id)
+        .then(() => {
+          setTodos(prevtodos => prevtodos.filter(task => task.id !== todo.id));
+        })
+        .catch(() => {
+          setErrorMessage(Errors.DeleteError);
+          setTimeout(() => {
+            setErrorMessage('');
+          }, 3000);
+        })
+        .finally(() => {
+          setMultiLoader(false);
+          setTimeout(() => {
+            if (titleField.current) {
+              titleField.current.focus();
+            }
+          }, 0);
```

Comment:
This logic can be moved into useEffect hook

Reply:
I don't know how do this. I don't know what to specify in the dependency array